### PR TITLE
Push master build to MyGet + validate Api Docs generation script

### PR DESCRIPTION
**Pull request by sanych-sun**
Push master build to MyGet
Validate Api Docs generation script (PR build probably has to be configured to include this variant)

### CODE REVIEW

File: evergreen/evergreen.yml [hunk 1]
```diff
@@ -1740,12 +1739,41 @@ tasks:
...
+        - func: download-package
+          vars:
+            PACKAGE_ID: "MongoDB.Driver.GridFS"
+        - func: download-package
+          vars:
+            PACKAGE_ID: "mongocsharpdriver"
+        - func: push-packages
+          vars:
+            PACKAGES_SOURCE: "https://www.myget.org/F/mongodb/api/v3/index.json"
+            PACKAGES_SOURCE_KEY: ${myget_api_key}
```
Comment (BorisDog) [inline evergreen/evergreen.yml:1763]: Polling would be additional func?

File: evergreen/evergreen.yml [hunk 2]
```diff
@@ -46,8 +46,11 @@ functions:
             CURRENT_VERSION=latest
           fi
 
-          if [ "${BUILD_TARGET}" = "release" ]; then
-            PACKAGE_VERSION=$(bash ./evergreen/get-version.sh)
+          PACKAGE_VERSION=$(bash ./evergreen/get-version.sh)
+          clear_version_rx='^[0-9]+\.[0-9]+\.[0-9]+(-[a-zA-Z0-9]+)?$'
+          if [ "${BUILD_TARGET}" = "release" ] && [[ ! "$PACKAGE_VERSION" =~ $clear_version_rx ]]; then
+            echo "Cannot run release script for non-clean version: '$PACKAGE_VERSION'"
```
Comment (BorisDog) [inline evergreen/evergreen.yml:52]:
```suggestion
            echo "Cannot run release script for dev version: '$PACKAGE_VERSION'"
```

**Review by BorisDog — CHANGES_REQUESTED** [review]

File: evergreen/evergreen.yml [hunk 3]
```diff
@@ -46,8 +46,11 @@ functions:
             CURRENT_VERSION=latest
           fi
 
-          if [ "${BUILD_TARGET}" = "release" ]; then
-            PACKAGE_VERSION=$(bash ./evergreen/get-version.sh)
+          PACKAGE_VERSION=$(bash ./evergreen/get-version.sh)
+          clear_version_rx='^[0-9]+\.[0-9]+\.[0-9]+(-[a-zA-Z0-9]+)?$'
+          if [ "${BUILD_TARGET}" = "release" ] && [[ ! "$PACKAGE_VERSION" =~ $clear_version_rx ]]; then
```
Comment (BorisDog) [inline evergreen/evergreen.yml:51]: Consider moving this check to `push-packages.sh `or somewhere in that area.

File: evergreen/evergreen.yml (hunk 1 above)
Comment (sanych-sun) [reply evergreen/evergreen.yml:1767]: Done, inside of `push-packages.sh`

File: evergreen/evergreen.yml [hunk 4]
```diff
@@ -1740,12 +1739,41 @@ tasks:
...
+            PACKAGE_ID: "MongoDB.Driver.GridFS"
+        - func: download-package
+          vars:
+            PACKAGE_ID: "mongocsharpdriver"
+        - func: push-packages
+          vars:
+            PACKAGES_SOURCE: "https://www.myget.org/F/mongodb/api/v3/index.json"
+            PACKAGES_SOURCE_KEY: ${myget_api_key}
+
+
```
[the remark this answers is not in the document]
Comment (sanych-sun) [reply evergreen/evergreen.yml]: Done

File: evergreen/evergreen.yml (hunk 3 above)
Comment (sanych-sun) [reply evergreen/evergreen.yml]: Done

**Review by BorisDog — APPROVED** [review]
LGTM

**Pull request merged**